collectors/python.d.plugin/zoneminder/zoneminder.chart.py

```diff
@@ -0,0 +1,188 @@
+# -*- coding: utf-8 -*-
+# Description: zoneminder netdata python.d module
+# Author: Jose Chapa
+# SPDX-License-Identifier: GPL-3.0-or-later
+# Zoneminder API: https://zoneminder.readthedocs.io/en/stable/api.html
+
+import time, os.path
+
+try:
+    import requests
+    import jwt
+    HAVE_DEPS = True
+except ImportError:
+    HAVE_DEPS = False
+
+from bases.FrameworkServices.SimpleService import SimpleService
+
+update_every = 10
+
+ORDER = [
+    'camera_fps',
+    'camera_bandwidth',
+    'events',
+    'disk_usage',
+]
+
+CHARTS = {
+    'camera_fps': {
+        'options': [None, 'Capture FPS', 'FPS', 'capture_fps', 'zm_camera.fps', 'line'],
+        'lines': []
+    },
+    'camera_bandwidth': {
+        'options': [None, 'Capture Bandwidth', 'kB/s', 'camera_bandwidth', 'zm_camera.bandwidth', 'stacked'],
+        'lines': []
+    },
+    'events': {
+        'options': [None, 'Events', 'count', 'events', 'zm_camera.events', 'stacked'],
+        'lines': []
+    },
+    'disk_usage': {
+        'options': [None, 'Disk Space', 'GB', 'disk_space', 'zm_camera.disk_space', 'area'],
+        'lines': [
+            ['zm_disk_space', 'used', 'absolute', None, 1073741824]
+        ]
+    },
+}
+
+def zm_generate_refresh_token(zoneminder_url, zm_user, zm_password, connection_timeout):
+    try:
+        post_data=dict()
+        post_data["user"] = zm_user
+        post_data["pass"] = zm_password
+        r = requests.post(zoneminder_url + '/api/host/login.json', data=post_data, timeout=connection_timeout, verify=False)
+        json_data = r.json()
+        if all (k in json_data for k in ("access_token","refresh_token")):
+            try:
+                token_file = open(os.path.expanduser("~/.zm_token.txt"),'w')
+                token_file.write("{}|{}".format(json_data["access_token"], json_data["refresh_token"]))
+                token_file.close()
+            except IOError:
+                return ("<error>", "Error while writing .zm_token.txt file.")
+            return ("ok", "{}|{}".format(json_data["access_token"], json_data["refresh_token"]))
+        return ("<error>", "Invalid api response when trying to generate new access and refresh tokens: " + r.text)
+    except requests.exceptions.RequestException as e:
+        return ("<error>", e)
+
+def zm_generate_access_token(zoneminder_url, refresh_token, connection_timeout):
+    try:
+        r = requests.post(zoneminder_url + '/api/host/login.json?token=' + refresh_token, timeout=connection_timeout, verify=False)
+        json_data = r.json()
+        if ("access_token" in json_data):
+            try:
+                token_file = open(os.path.expanduser("~/.zm_token.txt"),'w')
+                token_file.write("{}|{}".format(json_data["access_token"], refresh_token))
+                token_file.close()
+            except IOError:
+                return ("<error>", "Error while writing .zm_token.txt file.")
```

Comment:
You missed the directory that you are using when try to create the access token here, please, add `~/` in front of the file name.